src/reducers/application.js

```diff
@@ -55,7 +55,10 @@ export default (state = initialState, action) => {
     }
     case CLEAR_MESSAGE: {
       const messages = [...state.messages];
-      messages.pop(messages.map(message => message.key === action.payload.key));
+      messages.splice(
```

Comment:
Why is this change necessary? 🤔

Reply:
@lauriii The second parameter for `.splice()` is `deleteCount`. Previously it was omitted, meaning that all elements from the start to the end of the array were removed. Setting it to `1` indicates that we only want to remove one element.

Reply:
Also this is a switch from `.pop()` to `.splice()`.

Reply:
How about replacing all this code by ```return { ...state, messages: state.messages.filter(message => message.key !== action.payload.key)}```?

Reply:
The main motive to use splice() here is to **remove** an element from the state messages. Using a filter() method will also do the job, I believe.

Let me know if any changes are required :)